trie-db/src/node_codec.rs

```diff
@@ -19,12 +19,25 @@ use hash_db::Hasher;
 use node::Node;
 use ChildReference;
 
-use elastic_array::{ElasticArray128};
+#[cfg(feature = "std")]
+use std::error::Error;
+
+
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+
+use elastic_array::ElasticArray128;
+
+#[cfg(not(feature = "std"))]
+pub trait Error: core::fmt::Debug {}
```

Comment:
Requiring `Debug` on `no_std` does not sound that good.

Reply:
Yes it does not look super good, I did test, I enabled it to be able to use 'expect' in code (it is only 4 expect but if we do not implement it it seems super likely that we break no_std in next trie pr). I still can remove those 4 expect, what do you think?

Reply:
It is just that `Debug` can generate a lot of code that blows up the WASM size.